presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java

```diff
@@ -71,27 +70,13 @@ public class OrcBatchPageSource
 
     private final RuntimeStats runtimeStats;
 
-    private final List<Boolean> isRowPositionList;
+    private final List<Boolean> isRowNumberList;
```

Comment:
Do you mind converting it to IntOpenHashSet or boolean[]?

Reply:
sounds like a good idea. However, if I do that in this PR changes will slip into presto-iceberg and your approval alone is no longer sufficient, so perhaps that should be a separate change?

Reply:
You can convert List<Boolean> to the new type in the constructor and keep the interface untouched.